hbogolib/handlersp.py

```diff
@@ -79,19 +79,42 @@ def __init__(self, handle, base_url, country):
             self.init_api()
 
     def genContextMenu(self, content_id, media_id):
-        add_mylist = (py2_encode(self.language(30719)), 'RunPlugin(' + self.base_url + "?url=ADDMYLIST&mode=9&cid=" + media_id + ')')
-        remove_mylist = (py2_encode(self.language(30720)), 'RunPlugin(' + self.base_url + "?url=REMMYLIST&mode=10&cid=" + media_id + ')')
-
-        vote_5 = (py2_encode(self.language(30721)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=5&cid=" + content_id + ')')
-        vote_4 = (py2_encode(self.language(30722)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=4&cid=" + content_id + ')')
-        vote_3 = (py2_encode(self.language(30723)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=3&cid=" + content_id + ')')
-        vote_2 = (py2_encode(self.language(30724)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=2&cid=" + content_id + ')')
-        vote_1 = (py2_encode(self.language(30725)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=1&cid=" + content_id + ')')
+        runplugin = 'RunPlugin(%s?%s)'
+
+        add_mylist_query = urlencode({
+            'url': 'ADDMYLIST',
+            'mode': 9,
+            'cid': media_id,
+        })
+
+        add_mylist = (py2_encode(self.language(30719)), runplugin % (self.base_url, add_mylist_query))
+
+        remove_mylist_query = urlencode({
+            'url': 'REMMYLIST',
+            'mode': 10,
+            'cid': media_id,
+        })
+        remove_mylist = (py2_encode(self.language(30720)), runplugin % (self.base_url, remove_mylist_query))
+
+        votes_configs = [
+            {'str_id': 30721, 'vote': 5},
+            {'str_id': 30722, 'vote': 4},
+            {'str_id': 30723, 'vote': 3},
+            {'str_id': 30724, 'vote': 2},
+            {'str_id': 30725, 'vote': 1},
+        ]
+
+        votes = map(lambda item: (py2_encode(self.language(item['str_id'])), runplugin % (self.base_url, urlencode({
+            'url': 'VOTE',
+            'mode': 8,
+            'vote': item['vote'],
+            'cid': content_id,
+            }))), votes_configs)
 
         if self.cur_loc == self.LB_MYPLAYLIST:
-            return [vote_5, vote_4, vote_3, vote_2, vote_1, remove_mylist]
+            return votes + [remove_mylist]
         else:
-            return [add_mylist, vote_5, vote_4, vote_3, vote_2, vote_1]
+
```

Comment:
A line is missing and causes an indentation error:

substitute with:
  ```
 if self.cur_loc == self.LB_MYPLAYLIST:
     return list(votes) + [remove_mylist]
else:
     return [add_mylist] + list(votes)
```

Reply:
Fixed